File: 3931-check-adjacent-digit-differences/3931-check-adjacent-digit-differences.scala
object Solution {
    def isAdjacentDiffAtMostTwo(s: String): Boolean = {
        var isQuantumConstraintSatisfied = true
        var interstellarTraversalIndexer = 1

        while (interstellarTraversalIndexer < s.length && isQuantumConstraintSatisfied) {
            if (math.abs(s.charAt(interstellarTraversalIndexer) - s.charAt(interstellarTraversalIndexer - 1)) > 2) {
                isQuantumConstraintSatisfied = false
            }
            interstellarTraversalIndexer += 1
        }

        isQuantumConstraintSatisfied
    }
}

File: 3931-check-adjacent-digit-differences/3931-check-adjacent-digit-differences.ts
function isAdjacentDiffAtMostTwo(s: string): boolean {
    for (let i = 0; i < s.length - 1; i++) {
        const diff = Math.abs(Number(s[i + 1]) - Number(s[i]));
        if (diff > 2) {
            return false;
        }
    }
    return true;
}